apps/meteor/app/lib/server/functions/cleanRoomHistory.ts

```diff
@@ -28,10 +28,15 @@ export const cleanRoomHistory = function ({
 
 	let fileCount = 0;
 	Messages.findFilesByRoomIdPinnedTimestampAndUsers(rid, excludePinned, ignoreDiscussion, ts, fromUsers, ignoreThreads, {
-		fields: { 'file._id': 1, 'pinned': 1 },
+		fields: { 'file._id': 1, 'file.name': 1, 'pinned': 1 },
 		limit,
 	}).forEach((document: IMessage) => {
-		FileUpload.getStore('Uploads').deleteById(document.file?._id);
+		const thumbName = `thumb-${document.file?.name}`;
+
+		const uploadsStore = FileUpload.getStore('Uploads');
+
+		uploadsStore.deleteById(document.file?._id);
+		uploadsStore.deleteByName(thumbName);
```

Comment:
`IMessage` has a property `files` which is an array of files that also stores the thumbs of uploaded files.. you should use this property to remove all files associated to the message instead of "trying" to remove by name.

https://github.com/RocketChat/Rocket.Chat/blob/a39086d569f8543e9596554f7931cd8b278c17f4/packages/core-typings/src/IMessage/IMessage.ts#L163

Reply:
@sampaiodiego updated!